bundle/config/mutator/mutator.go

```diff
@@ -2,10 +2,13 @@ package mutator
 
 import (
 	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/config"
+	"github.com/databricks/cli/bundle/scripts"
 )
 
 func DefaultMutators() []bundle.Mutator {
 	return []bundle.Mutator{
+		scripts.Execute(config.ScriptPreInit),
```

Comment:
Note: this is not where this hook should be executed in the non-experimental version.

Before executing anything, we'd need to have already done:
1. target selection,
2. variable interpolation,
3. confirmed that our workspace is configured correctly.

Then to pick up new configuration (e.g. resources) from these scripts, we'd need a different mechanism. For example, we could set an environment variable `DATABRICKS_BUNDLE_RESOURCES` to some temporary file path that the script can write to, and we subsequently load + merge.

Given that this will be empty/a no-op for regular usage, I'm OK merging this and addressing this later.

Reply:
Agree with the above. Let's merge at current version as an experimental and iterate over it